### DISCUSSION TITLE
Using colors to enhance visualizations

### DISCUSSION THREAD

**Discussion by initialcommit-io** [initialcommit-com/git-sim#69] [General]
@abhijitnathwani @paketb0te Considering the visual nature of this tool, I think it's perfect for using colors to enhance the user experience and add additional information to the Git simulations where it makes sense.

For starters, I just released v0.2.7 which has the new global option `--color-by=author`.

This recolors the commit circles after drawing the graph based on the author of each commit and includes the color-coded author name in the frame along with the number of commits they authored:

Merge:

![git-sim-merge_03-06-23_22-27-16](https://user-images.githubusercontent.com/49353917/223351263-8a6550ac-fee5-495c-b4a3-088450876c49.jpg)

Rebase:

![git-sim-rebase_03-06-23_22-30-00](https://user-images.githubusercontent.com/49353917/223351331-18204a10-e9de-45d7-b06a-e8d62b583924.jpg)

Log Animation:

https://user-images.githubusercontent.com/49353917/223353418-ec4ae61b-24e8-42b4-a6ff-691b458ec66c.mp4

I can see us extending this to stuff like:

- coloring branches
- color commits by date range
- coloring commits that touch specific files
- etc

If you have any ideas for cool ways we can use color to create new Git-Sim features, would love to hear them =)

**Comment by abhijitnathwani** [comment]
This is cool!

**Comment by paketb0te** [comment]
Neat!

I think we can organize the coloring options nicely using an Enum, see #80 
This will serve two purposes,

a) make all the options visible when using `git-sim --help`
b) makes the string comparison in the `color_by()` function less error-prone, since most IDEs will immediately show an error when trying to use a nonexistent Enum member.

**Reply by initialcommit-io** [reply]
Smart.  Yes I agree let's do this!

Let me know if you can think of any other cool color-based features we could add into Git-Sim.
